Add support for clasp

drmeister opened the pull request:
Hi there - we have developed a new implementation of Common Lisp called Clasp
(https://github.com/clasp-developers/clasp.git).

Could you add this support for static vectors?

sionescu commented:
I see a few issues here:
- Please squash these commits into a single one
- Order things alphabetically in the .asd file
- static-vector-address is part of the API and as such it's required. I don't understand why it would be pointless on Clasp.

drmeister commented:
Hi - I think I fixed the things you asked for:
- I added static-vector-address (that was an oversight).
- I ordered the #+clasp feature test in the .asd file so it is in alphabetical order
- I squashed the commits into a single one

Thanks!

drmeister commented:
Hang on - I introduced a typo and didn't check it before pushing - fixing...

drmeister commented:
Ok, now I think everything is as you requested.

sionescu commented:
Pushed, thank you. Just to be sure, does the test suite pass all tests ?

kpoeck commented:
Test results for current clasp
```lisp
COMMON-LISP-USER> (5am:run! :static-vectors)

Running test suite STATIC-VECTORS
 Running test MAKE-STATIC-VECTOR.PLAIN.NOTINLINE .
 Running test WITH-STATIC-VECTOR.DEFAULTS .f../../src/gctools/gc_interface.cc:296 About to obj_deallocate_unmanaged_instance #1A(0 0 0 0 0 )

 Running test MAKE-STATIC-VECTOR.INITIAL-CONTENTS.NOTINLINE ..
 Running test WITH-STATIC-VECTOR.ELEMENT-TYPE .f../../src/gctools/gc_interface.cc:296 About to obj_deallocate_unmanaged_instance #1A(0 0 0 )

 Running test MAKE-STATIC-VECTOR.PLAIN.COMPILER-MACRO.NOERROR .
 Running test MAKE-STATIC-VECTOR.PLAIN.INLINE .
 Running test WITH-STATIC-VECTOR.INITIAL-CONTENTS ..../../src/gctools/gc_interface.cc:296 About to obj_deallocate_unmanaged_instance #1A(3 14 29 )

 Running test MAKE-STATIC-VECTOR.INITIAL-ELEMENT-AND-CONTENTS.COMPILER-MACRO.ERROR .
 Running test MAKE-STATIC-VECTOR.INITIAL-CONTENTS.COMPILER-MACRO.NOERROR .
 Running test MAKE-STATIC-VECTOR.INITIAL-CONTENTS.INLINE ..
 Running test MAKE-STATIC-VECTOR.INITIAL-ELEMENT.COMPILER-MACRO.NOERROR .
 Running test WITH-STATIC-VECTORS.INITIALIZATION ..../../src/gctools/gc_interface.cc:296 About to obj_deallocate_unmanaged_instance #1A(1 2 3 4 5 )
../../src/gctools/gc_interface.cc:296 About to obj_deallocate_unmanaged_instance #1A(7 7 7 )

 Running test MAKE-STATIC-VECTOR.INITIAL-ELEMENT.NOTINLINE ..
 Running test WITH-STATIC-VECTOR.INITIAL-ELEMENT ..../../src/gctools/gc_interface.cc:296 About to obj_deallocate_unmanaged_instance #1A(7 7 7 )

 Running test MAKE-STATIC-VECTOR.INITIAL-ELEMENT.INLINE ..
 Did 24 checks.
    Pass: 22 (91%)
    Skip: 0 ( 0%)
    Fail: 2 ( 8%)

 Failure Details:
 --------------------------------
 WITH-STATIC-VECTOR.ELEMENT-TYPE []: 
      
(ARRAY-ELEMENT-TYPE V)

 evaluated to 

EXT:BYTE16

 which is not 

EQUAL

 to 

(UNSIGNED-BYTE 16)

..
 --------------------------------
 --------------------------------
 WITH-STATIC-VECTOR.DEFAULTS []: 
      
(ARRAY-ELEMENT-TYPE V)

 evaluated to 

EXT:BYTE8

 which is not 

EQUAL

 to 

(UNSIGNED-BYTE 8)

..
 --------------------------------
NIL
(#<IT.BESE.FIVEAM::TEST-FAILURE> #<IT.BESE.FIVEAM::TEST-FAILURE>)
````

sionescu commented:
That looks like a bug in Clasp, that cl:array-element-type does not return canonical (standard) type names, assuming that EXT:BYTE8 is just an alias for (UNSIGNED-BYTE 8).

kpoeck commented:
I believe the code is in static-vectors, see "Pushed, thank you. "
I don't really understand why the pr is still open

sionescu commented:
Closing because the static-vectors side is done, but you should fix Clasp's cl:array-element-type otherwise static-vectors will not work properly.

kpoeck commented:
Thanks (and will check)

drmeister commented:
Thank you Stelian!


On Thu, Nov 28, 2019 at 1:25 AM Karsten Poeck <notifications@github.com>
wrote:

> Thanks (and will check)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/sionescu/static-vectors/pull/20?email_source=notifications&email_token=ABTVUVDLSH7QZDNE7ROI5O3QV5P4RA5CNFSM4G7KDX52YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFLSLOQ#issuecomment-559359418>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABTVUVFHZFSFIOTYDS7UUFDQV5P4RANCNFSM4G7KDX5Q>
> .
>


-- 
Christian Schafmeister
Professor, Chemistry Department
Temple University
